authlib/oauth2/rfc6749/grants/refresh_token.py

```diff
@@ -125,6 +126,9 @@ def create_token_response(self):
             scope = credential.get_scope()
 
         client = self.request.client
+        if client and client.client_id != credential.client_id:
```

Comment:
should probably be in `_validate_request_token` above, and return a generic invalid grant error to simulate a not found token